oci/spec_opts.go

```diff
@@ -522,6 +522,18 @@ func WithNamespacedCgroup() SpecOpts {
 func WithUser(userstr string) SpecOpts {
 	return func(ctx context.Context, client Client, c *containers.Container, s *Spec) error {
 		setProcess(s)
+
+		// For LCOW it's a bit harder to confirm that the user actually exists on the host as a rootfs isn't
+		// mounted on the host and shared into the guest, but rather the rootfs is constructed entirely in the
+		// guest itself. To accommodate this, a spot to place the user string provided by a client as-is is needed.
+		// The `Username` field on the runtime spec is marked by Platform as only for Windows, and in this case it
+		// *is* being set on a Windows host at least, but will be used as a temporary holding spot until the guest
+		// can use the string to perform these same operations to grab the uid:gid inside.
+		if s.Windows != nil && s.Linux != nil {
```

Comment:
Shouldn't it be 
if s.Windows != nil || s.Linux != nil ?? 

And if you include solaris as well form the spec then add that option as well?

Reply:
The common case for telling if a spec is for LCOW is if the Windows **and** Linux fields are filled in. This should only be the case (rather, the only normal case; a client can do whatever it wants I suppose) for LCOW. https://github.com/containerd/containerd/blob/f0a32c66dad1e9de716c9960af806105d691cd78/oci/spec.go#L71-L77

Do you know of another case where both of these would be filled in?

Reply:
@dcantah Thanks for the explanation! It was non-intuitive for me without that :)